### PR TITLE
Replaced synchronized String Object with Object

### PR DESCRIPTION
**Pull request by StefanSpieker**
I stumbled upon this when I was looking for SpotBugs issues. Actually spotbugs complained about the `private final Object logDirLock` which gets a string assigned. So I thought not optimizing the type but the whole locking. Is this somehow useful, or should I just change the object constructor?

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]
> Is this somehow useful

Useful? Likely not as useful as envisioned, because any _meaningful_ synchronization around the elements in the filesystem namespace is done at the VFS layer in the kernel. The difference between coarse and fine-grained locking at the userland layer is unlikely to matter at all in practice. One could make a subjective argument that the `synchronized` keyword version was simpler and more readable, but I do not place a high value on subjective arguments in code review. Therefore I am approving this change as-is.

**Review by oleg-nenashev — APPROVED** [review]
Whatever works w.r.t the lock object. TBH the implementation is a bit weird. The code always tries to set a lock and to create the directory, even if it already exists. I'd guess this is a subject from improvement, because the method will be called quite often.

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: core/src/main/java/hudson/model/Computer.java
Comment (basil) [inline core/src/main/java/hudson/model/Computer.java:199]:
```suggestion
    private final Object logDirLock = new Object();
```

**Review by basil — COMMENTED** [review]
@StefanSpieker The consensus seems to be that while this change isn't necessarily incorrect, it isn't necessarily the most straightforward solution either. Would you be willing to simplify this to use a regular object as the lock rather than a `ReentrantLock`? If so, then I think this would get a straightforward approval and merge without any further delay.

File: core/src/main/java/hudson/model/Computer.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/model/Computer.java]: Ok, will do so 👍

**Comment by StefanSpieker** [comment]
Please squash the commits then.

**Review by basil — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

**Comment by basil** [comment]
@oleg-nenashev Can this get a `ready-for-merge` label? I'd apply it myself, but so far I have been hesitant do that since I do not want to speak on behalf of the core team.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
> @oleg-nenashev Can this get a `ready-for-merge` label? I'd apply it myself, but so far I have been hesitant do that since I do not want to speak on behalf of the core team.

feel free to add, it means it shows up when we go to merge, normally I sanity check anyway before merging (if I didn't review it)


---

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by jglick — COMMENTED** [review]
I think the hope was that some future JVM would display monitors that happened to be `String`s specially in thread dumps.

**Pull request merged**
